compiler/src/iree/compiler/Dialect/Vulkan/Utils/TargetTriple.cpp

```diff
@@ -315,6 +317,8 @@ CapabilitiesAttr getCapabilities(const TargetTriple &triple,
 
       shaderFloat16 = shaderInt8 = shaderInt16 = true;
 
+      shaderIntegerDotProduct = true;
```

Comment:
Do we have plans to populate capabilities for triples in some more general way than this?

(e.g. share triple information with upstream MLIR, ask the Vulkan SDK, query some [local] database)

Reply:
IIRC @antiagainst explained that these triples are primarily for development purposes, not for 'deployment'. We don't have a concrete 'deployment' strategy for spirv/vulkan yet from what I know.

Reply:
Yes these are primarily for development and to make our own life easier. So we just keep a few triples for the case we daily work on right now. Proper deployment would mean building out the solutions for https://github.com/openxla/iree/issues/12230, and figure out the policies for dividing the complicated mobile landscape to different buckets of devices.